dask_mongo/core.py

```diff
@@ -115,19 +136,10 @@ def read_mongo(
     if not match:
         match = {}
 
-    with pymongo.MongoClient(appname=appname, **connection_kwargs) as mongo_client:
+    with get_client(connection_kwargs) as mongo_client:
         coll = mongo_client[database][collection]
 
-        nrows = next(
-            (
-                coll.aggregate(
-                    [
-                        {"$match": match},
-                        {"$count": "count"},
-                    ]
-                )
-            )
-        )["count"]
+        nrows = coll.count_documents(match)
```

Comment:
I suggest making this change in a separate PR since it's unrelated to caching clients. Note that using count_documents requires pymongo>=3.7 and currently this library has no lower bound on pymongo.